src/main/java/com/microsoft/aad/msal4j/AcquireTokenSilentSupplier.java

```diff
@@ -69,6 +70,14 @@ AuthenticationResult execute() throws Exception {
                 }
 
                 if (!StringHelper.isBlank(res.refreshToken())) {
+                    //There are certain scenarios where the cached authority may differ from the client app's authority,
+                    // such as when a request is instance aware. Unless overridden by SilentParameters.authorityUrl, the
+                    // cached authority should be used in the token refresh request
```

Comment:
this statement does not sound correct, cache can contain tokens issued by different authorities, for silent call, requested authority should be used